### ISSUE TITLE
Request is holded and also no log and exception when use version 0.9.3 

### ISSUE THREAD

**Issue by vicbell** [Nike-Inc/riposte#50]
I use this test code:
```java
public class HealthCheckEndpoint extends StandardEndpoint<Void, String> {

    private static final Matcher matcher = Matcher.match("/healthcheck");

    @Override
    public CompletableFuture<ResponseInfo<String>> execute(
        RequestInfo<Void> request, Executor longRunningTaskExecutor, ChannelHandlerContext ctx
    ) {
    	CompletableFuture<String> cf = supplyAsync(() -> {
    				try {
    					
    					// ***** Do some expensive computations, use sleep here instead ****
						Thread.sleep(1000);
					} catch (InterruptedException e) {
						// TODO Auto-generated catch block
						e.printStackTrace();
					}
	    			return "test";
    			}
    			, longRunningTaskExecutor
    			, ctx);
    	return cf.thenApply(this::doResponse);
    }
    
    private ResponseInfo<String> doResponse(String str) {
        return ResponseInfo.<String>newBuilder(str)
        		.withHttpStatusCode(200)
        		.build();
    }
    
	public <U> CompletableFuture<U> supplyAsync(Supplier<U> supplier, Executor longRunningTaskExecutor, ChannelHandlerContext ctx) {
		CompletableFuture<U> cf = CompletableFuture.supplyAsync(
				AsyncNettyHelper.supplierWithTracingAndMdc(
						supplier
						, ctx)
				, longRunningTaskExecutor);
		return cf;
	}
	
    @Override
    public Matcher requestMatcher() {
        return matcher;
    }
}
```
When i make a request by a brower(or make some more requests at the same time), the brower is always busy and no response. The server side does not print any log and exception, when i redo a request then some WARN log appears:
```java
2017-04-25T20:00:37,937+08:00 [epollEventLoopGroup-4-1] |-WARN  c.n.r.s.h.ChannelPipelineFinalizerHandler appname=datacenter environment=local version=NA traceId=785b3912f412b4f4 - The caller's channel was closed before a response could be sent. This means that an access log probably does not exist for this request. Distributed tracing will be completed now if it wasn't already done. Any dangling resources will be released. response_info_is_null=true
```
Ｗhen i go back to version 0.9.1 the issue go away, could you take a look at this?

**Comment by nicmunroe** [comment]
Thanks for the report @vicbell, unfortunately I'm unable to reproduce. I grabbed the latest [riposte microservice template](https://github.com/Nike-Inc/riposte-microservice-template) and replaced the healthcheck endpoint with the code you provided. I tried building (with `./gradlew clean build`) and running (with `./debugShadowJar.sh`) from the command line. I also tried building and running from IntelliJ. In all cases it worked fine with concurrent calls and from the browser (I tried both firefox and chrome).

I don't really have an explanation for you. Can you try with a fresh checkout of the riposte microservice template and just replace HealthCheckEndpoint with the code you provided here, built and run from the command line like I mention above? If it still doesn't work then I'd wonder if you have a VPN or local proxy on that might be interfering with traffic you're sending to localhost?

**Comment by vicbell** [comment]
I use netty 4.0.45.Final for some reason, I debug the code and finally find this issue is cause by the `OneTimeTask` is removed from netty 4.0.45.Final and the class `NonblockingEndpointExecutionHandler` (line 183) used `OneTimeTask`.So it throws `NoClassDefFoundError` and did not close the channel and neither print stacktrace message.
```java
    executor.execute(new OneTimeTask() {
             @Override
             public void run() {
                setResponseInfoAndActivatePipelineForResponse(state, responseInfo, ctx);
             }
         });
```
In Riposte 0.9.1 it did not use this class so it works fine. Finally i changed netty version  to 4.0.36.Final this issue go away.

**Comment by vicbell** [comment]
By the way, did you have any plan to upgrade Netty to 4.1 ?

**Comment by nicmunroe** [comment]
Ah! Thanks for continuing to dig and figure out what happened. That makes sense. I'll fix that code to not require `OneTimeTask` anymore.

Yes, we will update to the latest Netty eventually. We'll probably go to the latest Netty 4.0.x first, and then to the latest 4.1. We need to do some heavy stability and performance testing first though, so it may be a little while.

**Comment by vicbell** [comment]
That's great, thinks!

**Comment by nicmunroe** [comment]
@vicbell This should be fixed in version 0.9.4 that was just released. With that version of Riposte you should be able to upgrade the underlying Netty version and still have things work. If you do and run into other issues with newer versions of Netty please feel free to report them - we will be updating Riposte eventually so that could help us get a head start. Thanks!

**Comment by vicbell** [comment]
Wow, great work. ^_^
